### ISSUE TITLE
Tried to place key X at same level as Y

### ISSUE THREAD

**Issue by jwoertink** [sdogruyol/kemal#241]
I have these 3 routes
```crystal
get "/members/:id/edit" do |env|
end
get "/members/export" do |env|    
end
get "/members/:id/videos" do |env|
end
```

and they cause this error
```text
Tried to place key ':id/videos' at same level as ':id/edit' (Radix::Tree::SharedKeyError)
[4330602514] *CallStack::unwind:Array(Pointer(Void)) +82
[4330602417] *CallStack#initialize:Array(Pointer(Void)) +17
[4330602376] *CallStack::new:CallStack +40
[4330574217] *raise<Radix::Tree::SharedKeyError>:NoReturn +25
[4331284391] *Radix::Tree(Kemal::Route)@Radix::Tree(T)#add<String, Kemal::Route, Radix::Node(Kemal::Route)>:(Array(Radix::Node(Kemal::Route)) | Kemal::Route | Nil) +1223
[4331284427] *Radix::Tree(Kemal::Route)@Radix::Tree(T)#add<String, Kemal::Route, Radix::Node(Kemal::Route)>:(Array(Radix::Node(Kemal::Route)) | Kemal::Route | Nil) +1259
[4331283153] *Radix::Tree(Kemal::Route)@Radix::Tree(T)#add<String, Kemal::Route>:(Array(Radix::Node(Kemal::Route)) | Kemal::Route | Radix::Node(Kemal::Route) | Nil) +97
[4331282306] *Kemal::RouteHandler#add_to_radix_tree<String, String, Kemal::Route>:(Array(Radix::Node(Kemal::Route)) | Kemal::Route | Radix::Node(Kemal::Route) | Nil) +66
[4331282113] *Kemal::RouteHandler#add_route<String, String, &Proc(HTTP::Server::Context, Nil)>:(Array(Radix::Node(Kemal::Route)) | Kemal::Route | Radix::Node(Kemal::Route) | Nil) +65
[4330575095] *get<String, &Proc(HTTP::Server::Context, Nil)>:(Array(Radix::Node(Kemal::Route)) | Kemal::Route | Radix::Node(Kemal::Route) | Nil) +71
[4330495026] __crystal_main +4210
[4330556728] main +40
```

The interesting thing is that changing the order of these routes fixes it:
```crystal
get "/members/:id/edit" do |env|
end
get "/members/:id/videos" do |env|
end
get "/members/export" do |env|    
end
```
The tricky part is when you have 200 routes, and need to figure out the correct order ;) haha.

**Comment by sdogruyol** [comment]
I've never experienced a situation like this gues it's because of our `Radix` implementation 😢

**Comment by jwoertink** [comment]
Actually, the nice thing about it is that while porting over my rails app, I found a bug in my current routes 😂 It seems this is the winning order for routes

 # /
 # /:x
 # /:x/a
 # /:x/a/:y
 # /a/b
 # /a/b/:x

As long as I keep my routes in this order, my app will still boot. I have yet to test all of the routes to make sure they are doing what I want, but in the mini example above, this does work as expected.

**Comment by sdogruyol** [comment]
@jwoertink that's great to hear 😄

**Comment by luislavena** [comment]
@jwoertink both `:id/videos` and `:id/edit` should have been valid paths for the routing tree, so it appears to be a bug in [Radix](https://github.com/luislavena/radix). Feel free to submit a bug report or even better, a pull request :smile:

**Comment by jwoertink** [comment]
@luislavena thanks! I forked it and started working on a PR. I'd like to get your thoughts on direction since the tree parsing is slightly confusing 😂

**Comment by sdogruyol** [comment]
Closed in favor of https://github.com/luislavena/radix/issues/9

**Comment by luislavena** [comment]
@jwoertink @sdogruyol I've released Radix `v0.3.2` to address this bug. Please update to that release and let me know how it goes.

Thank you for your report!

:heart: :heart: :heart:

**Comment by sdogruyol** [comment]
@luislavena awesome! Now i can release Kemal 0.17.0 👍
